File: app/src/main/java/com/zaclimon/tsutaeru/util/NetworkUtils.java
package com.zaclimon.tsutaeru.util;

import java.io.BufferedInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.net.URL;
import java.net.URLConnection;

/**
 * Utility class for network related tasks
 *
 * @author zaclimon
 * Creation date: 13/08/17
 */

public class NetworkUtils {

    private static final int URLCONNECTION_CONNECTION_TIMEOUT_MS = 3000;  // 3 sec
    private static final int URLCONNECTION_READ_TIMEOUT_MS = 10000;  // 10 sec

    public static InputStream getNetworkInputStream(String url) throws IOException {

        URLConnection urlConnection = new URL(url).openConnection();
        urlConnection.setConnectTimeout(URLCONNECTION_CONNECTION_TIMEOUT_MS);
        urlConnection.setReadTimeout(URLCONNECTION_READ_TIMEOUT_MS);
        InputStream inputStream = urlConnection.getInputStream();

        return (inputStream == null ? null : new BufferedInputStream(inputStream));
    }

}

File: app/src/main/java/com/zaclimon/tsutaeru/util/NetworkUtils.kt
package com.zaclimon.tsutaeru.util

import java.io.InputStream
import java.net.URL

class NetworkUtils {

    companion object {
        // Connection timeout is 3 seconds
        private const val URLCONNECTION_CONNECTION_TIMEOUT_MS = 3000
        // Read timeout is 10 seconds
        private const val URLCONNECTION_READ_TIMEOUT_MS = 10000

        fun getNetworkInputStream(url: String): InputStream {
            val urlConnection = URL(url).openConnection().apply {
                connectTimeout = URLCONNECTION_CONNECTION_TIMEOUT_MS
                readTimeout = URLCONNECTION_READ_TIMEOUT_MS
            }
            val inputStream = urlConnection.getInputStream()
            return inputStream.buffered()
        }
    }

}